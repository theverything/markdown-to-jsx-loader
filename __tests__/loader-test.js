const path = require('path');
const fs = require('fs');
const deleteFile = require('rimraf').sync;
const webpack = require('webpack');

const outputFile = path.join(__dirname, 'fixtures/done.js');

function getWebpackConfig(opts) {
  const config = {
    target: 'web',
    entry: path.join(__dirname, 'fixtures/main.js'),
    output: {
      path: path.join(__dirname, 'fixtures'),
      filename: 'done.js',
    },
    module: {
      rules: [
        {
          test: /\.jsx.md$/,
          loader: path.resolve(__dirname, '../index.js'),
          options: opts || {},
        },
      ],
    },
    externals: {
      react: 'React',
    },
  };

  return config;
}

describe('markdown-to-jsx-loader', function () {
  afterEach(function () {
    deleteFile(outputFile);
  });

  it('wraps the markdown in div tag by default', function (done) {
    const compiler = webpack(getWebpackConfig());

    compiler.run(function (err, stats) {
      if (err) return done(err);

      const output = fs.readFileSync(outputFile, { encoding: 'utf8' });

      expect(output).toMatchSnapshot();

      done();
    });
  });

  it('wraps the markdown in span tag', function (done) {
    const compiler = webpack(getWebpackConfig({ wrapperTag: 'span' }));

    compiler.run(function (err, stats) {
      if (err) return done(err);

      const output = fs.readFileSync(outputFile, { encoding: 'utf8' });

      expect(output).toMatchSnapshot();

      done();
    });
  });

  it('returns a react pure function', function (done) {
    const compiler = webpack(getWebpackConfig({ wrapWith: 'function' }));

    compiler.run(function (err, stats) {
      if (err) return done(err);

      const output = fs.readFileSync(outputFile, { encoding: 'utf8' });

      expect(output).toMatchSnapshot();

      done();
    });
  });

  it('returns a react component', function (done) {
    const compiler = webpack(getWebpackConfig({ wrapWith: 'class' }));

    compiler.run(function (err, stats) {
      if (err) return done(err);

      const output = fs.readFileSync(outputFile, { encoding: 'utf8' });

      expect(output).toMatchSnapshot();

      done();
    });
  });
});
