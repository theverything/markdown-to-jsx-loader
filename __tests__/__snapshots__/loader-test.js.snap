// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown-to-jsx-loader returns a react component 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// identity function for calling harmony imports with the correct context
/******/ 	__webpack_require__.i = function(value) { return value; };
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 1);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const React = __webpack_require__(2);

module.exports = class test extends React.Component {
  render() {
    return React.createElement(
  \\"div\\",
  null,
  React.createElement(
    \\"h1\\",
    { id: \\"this-is-a-test\\" },
    \\"this is a test\\"
  ),
  React.createElement(
    \\"p\\",
    null,
    \\"hello world\\"
  ),
  React.createElement(
    \\"ul\\",
    null,
    React.createElement(
      \\"li\\",
      null,
      \\"hi\\"
    ),
    React.createElement(
      \\"li\\",
      null,
      \\"hell0\\"
    )
  )
);
  }
}

/***/ }),
/* 1 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
Object.defineProperty(__webpack_exports__, \\"__esModule\\", { value: true });
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md__ = __webpack_require__(0);
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md___default = __webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0__test_jsx_md__);



/***/ }),
/* 2 */
/***/ (function(module, exports) {

module.exports = React;

/***/ })
/******/ ]);"
`;

exports[`markdown-to-jsx-loader returns a react pure function 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// identity function for calling harmony imports with the correct context
/******/ 	__webpack_require__.i = function(value) { return value; };
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 1);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const React = __webpack_require__(2);

module.exports = function test() {
  return React.createElement(
  \\"div\\",
  null,
  React.createElement(
    \\"h1\\",
    { id: \\"this-is-a-test\\" },
    \\"this is a test\\"
  ),
  React.createElement(
    \\"p\\",
    null,
    \\"hello world\\"
  ),
  React.createElement(
    \\"ul\\",
    null,
    React.createElement(
      \\"li\\",
      null,
      \\"hi\\"
    ),
    React.createElement(
      \\"li\\",
      null,
      \\"hell0\\"
    )
  )
);
}

/***/ }),
/* 1 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
Object.defineProperty(__webpack_exports__, \\"__esModule\\", { value: true });
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md__ = __webpack_require__(0);
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md___default = __webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0__test_jsx_md__);



/***/ }),
/* 2 */
/***/ (function(module, exports) {

module.exports = React;

/***/ })
/******/ ]);"
`;

exports[`markdown-to-jsx-loader wraps the markdown in div tag by default 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// identity function for calling harmony imports with the correct context
/******/ 	__webpack_require__.i = function(value) { return value; };
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 1);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const React = __webpack_require__(2);

module.exports = React.createElement(
  \\"div\\",
  null,
  React.createElement(
    \\"h1\\",
    { id: \\"this-is-a-test\\" },
    \\"this is a test\\"
  ),
  React.createElement(
    \\"p\\",
    null,
    \\"hello world\\"
  ),
  React.createElement(
    \\"ul\\",
    null,
    React.createElement(
      \\"li\\",
      null,
      \\"hi\\"
    ),
    React.createElement(
      \\"li\\",
      null,
      \\"hell0\\"
    )
  )
);

/***/ }),
/* 1 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
Object.defineProperty(__webpack_exports__, \\"__esModule\\", { value: true });
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md__ = __webpack_require__(0);
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md___default = __webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0__test_jsx_md__);



/***/ }),
/* 2 */
/***/ (function(module, exports) {

module.exports = React;

/***/ })
/******/ ]);"
`;

exports[`markdown-to-jsx-loader wraps the markdown in span tag 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// identity function for calling harmony imports with the correct context
/******/ 	__webpack_require__.i = function(value) { return value; };
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 1);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

const React = __webpack_require__(2);

module.exports = React.createElement(
  \\"span\\",
  null,
  React.createElement(
    \\"h1\\",
    { id: \\"this-is-a-test\\" },
    \\"this is a test\\"
  ),
  React.createElement(
    \\"p\\",
    null,
    \\"hello world\\"
  ),
  React.createElement(
    \\"ul\\",
    null,
    React.createElement(
      \\"li\\",
      null,
      \\"hi\\"
    ),
    React.createElement(
      \\"li\\",
      null,
      \\"hell0\\"
    )
  )
);

/***/ }),
/* 1 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
Object.defineProperty(__webpack_exports__, \\"__esModule\\", { value: true });
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md__ = __webpack_require__(0);
/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__test_jsx_md___default = __webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0__test_jsx_md__);



/***/ }),
/* 2 */
/***/ (function(module, exports) {

module.exports = React;

/***/ })
/******/ ]);"
`;
